device_doctor/lib/src/android_device.dart

```diff
@@ -249,6 +256,51 @@ class AndroidDeviceDiscovery implements DeviceDiscovery {
     return healthCheckResult;
   }
 
+  /// The health check for battery level.
+  Future<HealthCheckResult> batteryLevelCheck({ProcessManager processManager}) async {
+    HealthCheckResult healthCheckResult;
+    try {
+      // The battery level returns two rows. For example:
+      //   level: 100
+      //   mod level: -1
+      final String levelResults = await eval('adb', <String>['shell', 'dumpsys', 'battery', '|', 'grep', 'level'],
+          processManager: processManager);
+      final List<String> levels = levelResults.trim().split('\n').toList();
```

Comment:
Can we replace all these lines with a regex?

Reply:
Replaced.